sequencer/src/state.rs

```diff
@@ -512,17 +543,71 @@ impl ValidatedState {
         } else {
             vec![]
         };
+        let accounts = std::iter::once(proposed_header.fee_info.account);
+
+        // Find missing state entries
+        let missing_accounts = self.forgotten_accounts(
+            accounts.chain(l1_deposits.iter().map(|fee_info| fee_info.account)),
+        );
+
+        let view = parent_leaf.get_view_number();
+
+        // Ensure merkle tree has frontier
+        if self.need_to_fetch_blocks_mt_frontier() {
+            tracing::warn!("fetching block frontier from peers");
+            instance
+                .peers
+                .as_ref()
```

Comment:
Wow just realized we need to replace all these `instance.peers` with `catchup`. Then we will pass in `instance.peers` as `catchup` _only_ in the case where we actually want to use peer nodes for catchup (ie when we're being called from HotShot

Reply:
_Or_, perhaps even better, we could get rid of the `catchup` argument, use `instance.peers`, and then _set_ `node_state.peers` to be the SQL catchup in `update_state_storage`